### PR TITLE
Fix Logout & reset-password

### PR DESCRIPTION
**Pull request by dmetzner**
Over the years some internal routes on share have changed. This PR should fix all routes that are accessed, but result in an 404 error.

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [ ] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Confirm that new and existing unit tests pass locally
- [ ] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/common/Constants.java [hunk 1]
```diff
@@ -141,7 +141,6 @@
 	public static final String CATROBAT_LOGIC_WIKI_URL = CATROBAT_FORMULA_WIKI_URL + "/Logic/";
 	public static final String CATROBAT_SENSORS_WIKI_URL = CATROBAT_FORMULA_WIKI_URL + "/Sensors/";
 	public static final String CATROBAT_OBJECT_WIKI_URL = CATROBAT_FORMULA_WIKI_URL + "/Properties/";
-	public static final String CATROBAT_TOKEN_LOGIN_URL = BASE_URL_HTTPS + "tokenlogin?username=";
```
Comment (dmetzner) [inline catroid/src/main/java/org/catrobat/catroid/common/Constants.java:144]: This route does not exist on Share.

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/dialog/login/LoginDialogFragment.java [hunk 2]
```diff
@@ -50,7 +50,7 @@
 public class LoginDialogFragment extends DialogFragment implements LoginTask.OnLoginListener {
 
 	public static final String TAG = LoginDialogFragment.class.getSimpleName();
-	public static final String PASSWORD_FORGOTTEN_PATH = "resetting/request";
+	public static final String PASSWORD_FORGOTTEN_PATH = "reset-password";
```
Comment (dmetzner) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/dialog/login/LoginDialogFragment.java:53]: We have a new route on share :)

File: catroid/src/main/java/org/catrobat/catroid/utils/Utils.java [hunk 3]
```diff
@@ -475,11 +474,8 @@ public static void invalidateLoginTokenIfUserRestricted(Context context) {
 
 	public static void logoutUser(Context context) {
 		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
-		String userName = sharedPreferences.getString(Constants.USERNAME, Constants.NO_USERNAME);
 		GoogleLoginHandler googleLoginHandler = new GoogleLoginHandler((AppCompatActivity) context);
 		googleLoginHandler.getGoogleSignInClient().signOut();
-		LogoutTask logoutTask = new LogoutTask(context, userName);
```
Comment (dmetzner) [inline catroid/src/main/java/org/catrobat/catroid/utils/Utils.java:481]: WebViewActivity.clearCookies(); already does everything that is necessary to be logged out in the webview.

**Review by juliajulie95 — APPROVED** [review]
Thank you

**Pull request merged**